accepted/2021/platform-dependencies/platform-dependencies.md

```diff
@@ -0,0 +1,433 @@
+
+* Docker: A Dockerfile that [installs](https://github.com/dotnet/dotnet-docker/blob/203e3e7c2cbb6a4ac73290c603ef02d249128500/src/runtime-deps/3.1/buster-slim/amd64/Dockerfile#L7-L14) dependent packages
+* Linux packages:  A deb package that [lists](https://github.com/dotnet/installer/blob/7f8f4a894abc5f666c2bfa45ce6e7c6fe9ace71b/src/redist/targets/packaging/deb/dotnet-debian_config.json#L31-L38) the packages it depends on.
+* Docs: User documentation that [lists](https://github.com/dotnet/docs/blob/b9a09326817f9acf4dab825f62ae0a3fb2427a30/docs/core/install/linux-debian.md#dependencies) the Linux packages that are necessary to have installed for .NET.
+
+They want to ensure this list is up-to-date with .NET's requirements. They author a tool which reads from .NET's platform dependency model and compares it against their package list to ensure it aligns for a given platform and version.
+
+### Determine vulnerability exposure
+
+A security incident has been disclosed in a Linux package that could potentially affect a platform dependency that .NET has. The response team needs to have an understanding of what components of .NET have a dependency on the Linux package so that the appropriate mitigation can be implemented. They search through .NET's platform dependency model to determine which .NET components are affected and on which platforms.
+
+### Dependency-tracking tools
+
+The .NET team wants to provide users with self-service tools so that they can know, a priori, what platform dependencies their specific application has. This allows them to configure the systems that will host their application with confidence. The developers creating these tools need a single source of truth from which to derive .NET's platform dependencies. They author the tools to read from the platform dependency model to get this information.
+
+### Dependency change alerting
+
+A .NET contributor makes a change to the NativeMethods.cs file in their project. They create a PR with their changes. A bot alerts the contributor to the potential need to update .NET's platform dependency model in response to the change in the NativeMethods.cs file. The bot alerts them by adding a special label to the PR and a comment that provides instructions on next steps they should follow.
+
+### Platform EOL
+
+Debian 9 goes EOL and a maintainer of the platform dependency model needs to clean up the model to remove references to Debian 9 since those references are now obsolete. They run a CLI command to update the model automatically:
+
+```console
+dotnet-deps platform remove debian.9
+```
+
+## Requirements
+
+### Goals
+
+* Common schema capable of describing both runtime and toolchain dependencies.
+* Model that limits repetition to make maintenance easier.
+* File format that is machine-readable to allow for automated transformation into other output formats.
+* Ability to describe:
+    * Dependencies for multiple platform types (Linux, Windows, MacOS)
```

Comment:
What about more specifics in that platform? 

For example, .NET Core 2.0 only supported using OpenSSL 1.0.x. .NET Core 2.1 got additional fixes that let it also use OpenSSL 1.1.y. It would be great it we could express things like that in this manifest: `needs-one-of (OpenSSL 1.0, OpenSSL 1.1)`.

I am also wondering what happens across builds. A portable build of .NET 5 SDK that bundles dependencies will have fewer build and runtime requirements than a non-portable build of .NET 5 SDK. The non-portable build will, for example, need the matching version of OpenSSL.

Reply:
> For example, .NET Core 2.0 only supported using OpenSSL 1.0.x. .NET Core 2.1 got additional fixes that let it also use OpenSSL 1.1.y. It would be great it we could express things like that in this manifest: needs-one-of (OpenSSL 1.0, OpenSSL 1.1).

I've redesigned the schema to support this.  Dependency names can now be described as an expression with logical OR operators.  Take a look: https://github.com/dotnet/designs/pull/184/commits/b7833d0705cfcaa9bc23f5d3024099056c492fe0

> I am also wondering what happens across builds. A portable build of .NET 5 SDK that bundles dependencies will have fewer build and runtime requirements than a non-portable build of .NET 5 SDK. The non-portable build will, for example, need the matching version of OpenSSL.

Since that would have been done by a third party and customized to suit their desired configuration, it's not really possible to describe that here, nor is it really relevant. The intent is to describe the dependencies of the assets distributed by Microsoft.  And as I mention [here](https://github.com/dotnet/designs/pull/184#discussion_r596927235), the model description has no impact on the functionality of .NET that has been built/distributed external of Microsoft.